### ISSUE TITLE
Avoid dollar in function and variable names?

### ISSUE THREAD

**Issue by plajjan** [actonlang/acton#167]
We use the dollar sign in function and variable names. I recently ran into problem trying to use our naming in Rust (writing an Acton module in Rust) and then looked a little deeper. Turns out some platforms do not support this either due to the assembler not accepting them - https://gcc.gnu.org/onlinedocs/gcc/Dollar-Signs.html#Dollar-Signs

Should we use some other character or come up with a different naming scheme?

@nordlander thoughts?

**Comment by nordlander** [comment]
Somehow we must be able to separate the Acton namespace from that of C, so that we can compile an Acton function called `printf` (say) and have it end up as something entirely separate from the C stdio function. When I discovered that most C compilers de facto support the '$' character in identifier names I thought it offered a great way of constructing C names in the rts that have no way of clashing with any names an Acton programmer might produce.

So the '$' fills a purpose. However, I much later realized that this approach only makes sense if the overarching goal really is to preserve Acton's _local_ variable names in the generated C code. For global names that goal is unfeasible anyway, since they must be mangled with module prefixes in order to guarantee uniqueness (for which we also use '$'). And preserving local names isn't that important after all, at least not more important than preserving top-level ones.

So I've come to realize that of course there exists a better naming scheme, which also happens not to depend on the '$' character. Roughly, one could pick a prefix like "A_" that would be prepended to _all_ translated Acton identifiers, even those that happen to start with "A_" themselves (which can happen, since "A_" is a legal name prefix in Acton). But since this would be done to all identifiers, there would be no risk of confusion as long as the rts and builtin library refrains from defining names that start with "A_".

The actual scenario in the compiler is a little more complicated, though, because one also has to uniquely encode the module prefixes of top-level names and also provide a namespace for identifier _variants_ (like the typedef, method-table and constructor function names of a single class, for example). But all this ought to be possible to mix with the general idea of escaping the translated Acton names instead of the C names of the execution environment.

So in short I think we should switch to such a new naming scheme at some point. But perhaps it isn't of the highest priority right now.

**Comment by plajjan** [comment]
Alright, excellent. Yes, I agree, not a high priority. I just happened to run into this and figured that if we could adopt a different naming, we could easily avoid the problem. This isn't hard, just tedious. We'll get around to it sometime!

**Comment by plajjan** [comment]
Closing in preference of #1042.
